sycl/test/check_device_code/group_barrier.cpp

```diff
@@ -0,0 +1,58 @@
+// RUN: %clangxx -fsycl-device-only -fsycl-unnamed-lambda -S -Xclang -emit-llvm %s -o - | FileCheck %s
+#include <sycl/sycl.hpp>
+
+const auto TestLambda = [](auto G) {
+  sycl::group_barrier(G);
+  sycl::group_barrier(G, sycl::memory_scope_work_item);
+  sycl::group_barrier(G, sycl::memory_scope_sub_group);
+  sycl::group_barrier(G, sycl::memory_scope_work_group);
+  sycl::group_barrier(G, sycl::memory_scope_device);
+  sycl::group_barrier(G, sycl::memory_scope_system);
+};
+
+int main() {
+  sycl::queue Q;
+
+  Q.submit([](sycl::handler &CGH) {
+    CGH.parallel_for(sycl::nd_range{sycl::range{1}, sycl::range{1}},
+                     [](sycl::nd_item<1> item) {
+                       auto G = item.get_group();
+                       TestLambda(G);
+                     });
+  });
+  Q.submit([](sycl::handler &CGH) {
+    CGH.parallel_for(sycl::nd_range{sycl::range{1, 1}, sycl::range{1, 1}},
+                     [](sycl::nd_item<2> item) {
+                       auto G = item.get_group();
+                       TestLambda(G);
+                     });
+  });
+  Q.submit([](sycl::handler &CGH) {
+    CGH.parallel_for(sycl::nd_range{sycl::range{1, 1, 1}, sycl::range{1, 1, 1}},
+                     [](sycl::nd_item<3> item) {
+                       auto G = item.get_group();
```

Comment:
Could you please add a test with  sub_group's?

Reply:
There's already a test in llvm-test-suite. But I can add one more.

Reply:
It would be great.